### PR TITLE
Generalise PyCharm Remote Debugger Instructions

### PR DESCRIPTION
**Pull request by cailafinn**
**Description of work.**
- Generalised instructions for using the PyCharm remote debugger that ships with the professional edition. 

**To test:**
- Check docs build correctly and look good. 

*There is no associated issue.*

*This does not require release notes* because **it only concerns developer documentation**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by martyngigg — CHANGES_REQUESTED** [review]
This looks good, thanks for improving it. I know these aren't in sections you changed but could you update the references to python 2.7 that are left:

* `/usr/bin/python2.7` --> `/usr/bin/python3`
* `<Mantid Source Directory>/external/src/ThirdParty/lib/python2.7/python.exe` -> `<Mantid Source Directory>/external/src/ThirdParty/lib/python3.8/python.exe`

File: dev-docs/source/PyCharm.rst [hunk 1]
```diff
@@ -23,7 +23,7 @@ Setting up PyCharm on Windows
 
    .. code-block:: sh
 
-      <Mantid Source Directory>/external/src/ThirdParty/lib/python2.7/python.exe
+      <Mantid Source Directory>/external/src/ThirdParty/lib/python3/python.exe
```
Comment (martyngigg) [inline dev-docs/source/PyCharm.rst:26]: This isn't quite right for windows as the directory is specifically `python3.8`

File: dev-docs/source/PyCharm.rst [hunk 2]
```diff
@@ -103,7 +103,7 @@ This can be done in two ways:
        <Mantid Build Directory>\bin\Debug;
        <Mantid Source Directory>\external\src\ThirdParty\bin;
        <Mantid Source Directory>\external\src\ThirdParty\bin\mingw;
-       <Mantid Source Directory>\external\src\ThirdParty\lib\python2.7;
+       <Mantid Source Directory>\external\src\ThirdParty\lib\python3;
```
Comment (martyngigg) [inline dev-docs/source/PyCharm.rst:106]: Similar to above

File: dev-docs/source/PyCharm.rst [hunk 3]
```diff
@@ -190,7 +190,7 @@ To use the remote debugger:
 Setting up PyCharm on Linux
 ###########################
 
-1. Use the native python interpreter (``/usr/bin/python2.7``) rather than from ``<Mantid Source Directory>/external/src/ThirdParty/lib/python2.7/python.exe``
+1. Use the native python interpreter (``/usr/bin/python3``) rather than from ``<Mantid Source Directory>/external/src/ThirdParty/lib/python3/python.exe``
```
Comment (martyngigg) [inline dev-docs/source/PyCharm.rst:193]: `/usr/bin/python3` is correct but the latter link needs correcting the same as above.

**Review by martyngigg — CHANGES_REQUESTED** [review]

File: dev-docs/source/PyCharm.rst (hunk 1 above)
Comment (ConorMFinn) [reply dev-docs/source/PyCharm.rst:26]: Don't we use pyhton3.7 on ubuntu at the moment, though?

**Review by martyngigg — APPROVED** [review]
Great! LGTM :shipit:

**Pull request merged**